ament_cmake_core/cmake/core/all.cmake

```diff
@@ -53,12 +53,23 @@ _set_cmake_policy_to_new_if_available(CMP0020)
 _set_cmake_policy_to_new_if_available(CMP0021)
 _set_cmake_policy_to_new_if_available(CMP0022)
 _set_cmake_policy_to_new_if_available(CMP0023)
+_set_cmake_policy_to_new_if_available(CMP0042)
```

Comment:
This policy is only available as of CMake 3.0. We currently rely on 2.8.x only. I am not sure that it is a good idea to set "newer" policies since it will result in different behavior depending on the CMake version used. May be we should consider to raise the minimum requirements of CMake?

On the other hand the patch sets `MACOSX_RPATH` explicitly to `ON` so we don't need this policy at all, right?